integrations/wiz/main.py

```diff
@@ -26,31 +29,49 @@ def init_client() -> WizClient:
     )
 
 
-@ocean.on_resync()
-async def on_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+@ocean.on_resync(ObjectKind.PROJECT)
+async def resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
     wiz_client = init_client()
 
-    if kind == ObjectKind.PROJECT:
-        async for projects in wiz_client.get_projects():
-            logger.info(f"Received {len(projects)} projects")
-            yield projects
-    else:
-        async for _issues in wiz_client.get_issues():
-            logger.info(f"Received {len(_issues)} issues")
-            if kind == ObjectKind.ISSUE:
-                yield _issues
-            elif kind == ObjectKind.CONTROL:
-                yield [
-                    issue["sourceRule"]
-                    for issue in _issues
-                    if issue.get("sourceRule") is not None
-                ]
-            elif kind == ObjectKind.SERVICE_TICKET:
-                yield [
-                    ticket
-                    for issue in _issues
-                    for ticket in issue.get("serviceTickets", [])
-                ]
+    async for projects in wiz_client.get_projects():
+        logger.info(f"Received {len(projects)} projects")
+        yield projects
+
+
+@ocean.on_resync(ObjectKind.ISSUE)
+async def resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    wiz_client = init_client()
+
+    status_list = typing.cast(
+        IssueResourceConfig, event.resource_config
+    ).selector.status_list
+
+    logger.info(f"Resyncing issues with status list: {status_list}")
+
+    async for issues in wiz_client.get_issues(status_list):
+        yield issues
+
+
+@ocean.on_resync(ObjectKind.CONTROL)
+async def resync_controls(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    wiz_client = init_client()
+
+    async for _issues in wiz_client.get_cached_issues():
```

Comment:
this will not work if user didn't choose to map kind `issue`, lets not cache

Reply:
I have changed to the default issues method with default status list of ["OPEN", "IN_PROGRESS"]